Enabling running scripts per configuration per framework

moozzyk opened the pull request:
Required to fix #887

davidfowl commented on src/Microsoft.Dnx.Tooling/Building/BuildManager.cs:
```diff
@@ -127,192 +125,208 @@ private bool BuildInternal(string projectPath)
+            return success;
+        }
 
+        private bool BuildConfiguration(string baseOutputPath, IEnumerable<System.Runtime.Versioning.FrameworkName> frameworks, List<DiagnosticMessage> allDiagnostics, string configuration)
```
Fix `System.Runtime.Versioning.FrameworkName` that. Fix the namespace.

moozzyk replied:
= don't trust "Extract Method" by VS. Will do.

davidfowl commented:
This diff is crazy hard to return

davidfowl commented:
We need to flow the context to the command being run. You need to add some variables to `GetScriptVariable` so that scripts can use `project:Configuration` and `project:Framework` when executing scripts to make decisions.

moozzyk commented:
@davidfowl d@sdfjeuio

moozzyk commented:
@davidfowl as I said it is required to fix #887 and not fixes #887. The fix prototype for bug-887 is here https://github.com/aspnet/dnx/commit/69b90084c6703c03ca7edd96d784e86316f901e1. I decided to send this separately since I am waiting for #2643. Without #2643 I don't have framework moniker - only the full name of framework. Also in the prototype I used different names - let me know which ones you want.

moozzyk commented:
The method was already like 300 lines long - I extracted a couple methods and simplified some `if`s.  It's hard to review but it is now easier to read.

davidfowl commented:
I think we should just pass the variables in the same time. How does this affect the performance of building a single project in our repository (since it now copies multiple times instead of once)?. I think the `build:TargetFramework` and `build:Configuration` sound better.

moozzyk commented:
I will wait for @victorhurdugaci and update the PR after he commits #2643. 
I have not checked the performance but from what I understand currently we build only one configuration and then copy explicitly for dnx451 and dnxcore50 (btw. this is also conceptually wrong since it assumes we always build both dnx451 and dnxcore50 which does not have to be the case). After the change the scripts need to/will be changed to copy only once but to %build.Configuration%/%build.TargetFramework% folder. So instead of doing all the copying blindly without really knowing what was built each build will take care of stuff it built.

moozzyk commented:
Replaced by #2679
